app/components/channel-nav/template.hbs

```diff
@@ -1,23 +1,51 @@
 <ul {{did-insert this.bindKeyboardShortcuts}}
     {{will-destroy this.unbindKeyboardShortcuts}}>
   {{#each @spaces as |space|}}
-    <li>
-      <h2><LinkTo @route="space" @model={{space}}>{{space.name}}</LinkTo></h2>
-      <ul>
-        {{#each space.sortedChannels as |channel|}}
-          <li class="{{if channel.connected "connected" "disconnected"}} {{channel.unreadMessagesClass}} {{if channel.visible "active" ""}}">
-            {{#if channel.isUserChannel}}
-              <LinkTo @route="space.user-channel" @models={{array space channel}}>{{channel.name}}</LinkTo>
-            {{else}}
-              <LinkTo @route="space.channel" @models={{array space channel}}>{{channel.name}}</LinkTo>
-            {{/if}}
-            <a {{on "click" (fn @onLeaveChannel space channel)}} class="leave-channel" title="Leave {{channel.name}}">x</a>
+    {{#if (eq space.protocol "IRC")}}
+      <li>
+        <h2>
+          <LinkTo @route="space" @model={{space}}>{{space.name}}</LinkTo>
+        </h2>
+        <ul>
+          {{#each space.sortedChannels as |channel|}}
+            <li class="{{if channel.connected "connected" "disconnected"}} {{channel.unreadMessagesClass}} {{if channel.visible "active" ""}}">
```

Comment:
Maybe we can extract this `<li>` into a subcomponent `channel-nav/channel` as it seems to be the same for `XMPP`.

Reply:
Thanks! I'll leave it for now, as it does not regard the actual improvements, and the changes also don't add much technical debt IMO. Refactors in separate PRs are obviously invited and encouraged. ;)